projects/storefrontlib/shared/services/import-export/export-csv.service.ts

```diff
@@ -0,0 +1,48 @@
+import { Injectable } from '@angular/core';
+import { FileOptions } from '../../models/file';
+import { ExportService } from './export.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ExportCsvService {
+  constructor(protected exportService: ExportService) {}
+  /**
+   * Converts array of objects into CSV data structure.
+   *
+   * @param objectsArray Array of objects which should be converted to CSV.
+   * @param separator for csv data
+   * @returns Processed string ready to be saved into file
+   */
+  dataToCsv(objectsArray: string[][], separator: string): string {
+    const array =
+      typeof objectsArray != 'object' ? JSON.parse(objectsArray) : objectsArray;
+    return array.reduce((csvString: string, row: string[]) => {
+      const line = row.reduce((currentLine, column) => {
+        currentLine += currentLine !== '' ? separator : '';
+        const cell = column.includes(separator) ? `"${column}"` : column;
+        return `${currentLine}${cell}`;
+      }, '');
+      return `${csvString}${line}\r\n`;
+    }, '');
+  }
+
+  /**
+   * Downloads CSV file
+   *
+   * @param objectsArray Array of objects which should be converted to CSV.
+   * @param separator for csv data
+   * @param fileOptions FileOptions
+   * @returns Processed string ready to be saved into file
+   */
+  downloadCsv(
```

Comment:
nitpick: it's already csv-related service. No need to repeat the context in the method name. What about `ExportCsvFileService.download()`?